File: src/main/java/com/hjj/apiserver/repositroy/PurchaseRepositoryCustom.java
package com.hjj.apiserver.repositroy;

import com.hjj.apiserver.domain.purchase.Purchase;
import com.hjj.apiserver.dto.PurchaseDto;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface PurchaseRepositoryCustom {

    PurchaseDto.ResponsePurchaseDetail findPurchase(Long userNo, Long purchaseNo);
    List<Purchase> findPurchasePageCustom(PurchaseDto purchaseDto, Pageable pageable);
}

File: financial/src/main/kotlin/ninja/sundry/financial/repository/purchase/PurchaseRepositoryCustom.kt
package com.hjj.apiserver.repository.purchase

import com.hjj.apiserver.domain.purchase.Purchase
import com.hjj.apiserver.dto.purchase.response.PurchaseDetailResponse
import org.springframework.data.domain.Pageable
import java.time.LocalDate

interface PurchaseRepositoryCustom {
    fun findPurchase(
        userNo: Long,
        purchaseNo: Long,
    ): PurchaseDetailResponse?

    fun findPurchasePageCustom(
        searchStartDate: LocalDate,
        searchEndDate: LocalDate,
        accountBookNo: Long,
        pageable: Pageable,
    ): List<Purchase>
}
